stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -10,19 +10,147 @@
 package com.facebook.stetho.inspector.elements.android;
 
 import android.view.View;
+import android.view.ViewDebug;
+import com.facebook.stetho.common.LogUtil;
 import com.facebook.stetho.common.StringUtil;
 import com.facebook.stetho.common.android.ResourcesUtil;
-import com.facebook.stetho.inspector.elements.AttributeAccumulator;
 import com.facebook.stetho.inspector.elements.AbstractChainedDescriptor;
+import com.facebook.stetho.inspector.elements.AttributeAccumulator;
+import com.facebook.stetho.inspector.elements.StyleAccumulator;
 
 import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
 
 final class ViewDescriptor extends AbstractChainedDescriptor<View> implements HighlightableDescriptor {
   private static final String ID_ATTRIBUTE_NAME = "id";
-
   private final MethodInvoker mMethodInvoker;
 
+  /**
+  * NOTE: Only access this via {@link #getCapitalPattern}.
+  */
+  private @Nullable Pattern mCapitalPattern;
+
+  /**
+   * NOTE: Only access this via {@link #getViewGetters}.
+   */
+  @Nullable
+  @GuardedBy("this")
+  private volatile Set<Method> VIEW_GETTERS;
+
+  private Pattern getCapitalPattern() {
+    if (mCapitalPattern == null) {
+      mCapitalPattern = Pattern.compile("(?=\\p{Upper})");
+    }
+
+    return mCapitalPattern;
+  }
+
+  private final Set<Method> getViewGetters() {
+    if (VIEW_GETTERS == null) {
+      synchronized (this) {
+        if (VIEW_GETTERS == null) {
+          Set<String> allowedGetters = new HashSet<>(
+              Arrays.asList(
+                  "getHorizontalScrollFactor",
```

Comment:
we should alphabetize this list too, otherwise it's really difficult to navigate